### ISSUE TITLE
Clarification question for `show()` algorithm

### ISSUE THREAD

**Issue by rsolomakhin** [w3c/browser-payment-api#820]
**TL;DR:** Can `show()` be called on the same instance of `PaymentRequest` multiple times?

The `show()` algorithm includes the following steps:

> 6. If *request*.[[state]] is not "created" then return a promise rejected with an "InvalidStateError" DOMException.
> 7. If the user agent's "payment request is showing" boolean is true, then return a promise rejected with an "AbortError" DOMException.
> 8. Set *request*.[[state]] to "interactive".

It looks like `show()` can be rejected without altering the *request*.[[state]], so the merchant can call `show()` on the same instance of `PaymentRequest`. Is this the only edge case where `show()` does not change state? Or am I reading the spec wrong?

**Comment by marcoscaceres** [comment]
> TL;DR: Can show() be called on the same instance of PaymentRequest multiple times?

So, yes:

 * Step 6 prevents "recycling" a payment request. 
 * But, step 7 does leave the request in a "created" state. 

I don't think that's a bug... it's an oddity, for sure... but I don't think it's terrible. 

I'll check what Firefox does...

**Comment by marcoscaceres** [comment]
Ok, made a fiddle:  https://jsfiddle.net/cuz5mt6h/11/

* Chrome "InvalidStateError"
* Firefox "InvalidStateError".

Ok, so, in step 7, we should set the state to "closed". 

@aestes, is that ok?

**Comment by marcoscaceres** [comment]
Ok, made a test page for Safari:

https://fiddle.jshell.net/cuz5mt6h/11/show/

Safari follows the spec (it `show()`s the second request, even after aborting the first `.show()`). I don't have a strong opinion - so if @aestes is ok with us changing step 7 above to "close" the state (and there is no valid use case for it), then that's fine.

I'll draft a PR in the meantime, as it's really simple to fix.
